### ISSUE TITLE
ps_venn and ps_euler -- obtain list of shared and not-shared OTUs?

### ISSUE THREAD

**Issue by reige012** [Russel88/MicEco#4]
Hi, I'm loving the ps_venn and ps_euler commands. They make great images and are super easy to use, but is there a way to also output the list of shared and not-shared ASVs or OTUs? I'd like to be able to cross-reference the data to some other things I'm doing. I've looked over the readme files, but I'm not sure savvy with code so I couldn't find an apparent way to easily do this.

Thanks!
Alicia Reigel

**Comment by Russel88** [comment]
Dear Alicia,

I'm glad you find it useful. 
It is not possible to get a list of shared and non-shared taxa, as the function is now. That part of the calculation is done in the `eulerr` package, so it's not that simple to incorporate it into this function unfortunately.

You could do something like this to help you get that information (`ps` is your phyloseq, `group` is your grouping factor, `fraction` is similar to as descriped in the `ps_venn()` help):
```
ps <- transform_sample_counts(ps, function(x) x/sum(x))
if(taxa_are_rows(ps)){
    ps_melted <- reshape2::melt(otu_table(ps))
} else {
    ps_melted <- reshape2::melt(t(otu_table(ps)))
}
ps_melted <- merge(ps_melted, sample_data(ps), by.x = "Var2", by.y = "row.names")
ps_agg <- aggregate(as.formula(paste("value ~ Var1 +",group)), data = ps_melted, function(x) (sum(x > 0)/length(x) >= fraction) * mean(x))
ps_mat <- reshape2::dcast(as.formula(paste("Var1 ~ ",group)), data = ps_agg, value.var = "value")
```

Then `ps_mat` contains a line per taxa, and a column per group. The number is the average relative abundance of that taxa in that group. From that data.frame you can then extract those that are shared and those that are unique to a group

**Comment by Russel88** [comment]
Dear Alicia,

If you still need this feature, it is now implemented in the latest MicEco version (0.9.14). Just run ps_venn or ps_euler with plot=FALSE and you will get a list of shared and unique taxa.

Best regards,
Jakob
